aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -607,6 +617,80 @@ namespace {
 
 } // anonymous namespace
 
+// Utilities exposed in RNNUtils.h
+namespace cudnn_rnn {
+
+  TORCH_CUDA_API std::tuple<Tensor, std::vector<Tensor>> copy_weights_to_flat_buf_views(
```

Comment:
Any substantive change to logic here?

Reply:
Added knobs are to make sure it can service both [_cudnn_rnn_flatten_weight](https://github.com/pytorch/pytorch/pull/42385/files#diff-85792c9e6600ba236d0db7382f29108fR698) and [autocast::_cudnn_rnn_cast_reflatten](https://github.com/pytorch/pytorch/pull/42385/files#diff-9ef93b6a4fb5a06a37c562b83737ac6aR336).

The existing behavior of `_cudnn_rnn_flatten_weight` should remain unchanged.